ParkingZoneApp/Repositories/Repository.cs

```diff
@@ -0,0 +1,45 @@
+﻿
+using Microsoft.EntityFrameworkCore;
+using ParkingZoneApp.Data;
+
+namespace ParkingZoneApp.Repositories
+{
+    public class Repository<T> : IRepository<T> where T : class
+    {
+        private readonly ApplicationDbContext _context;
+        private readonly DbSet<T> _dbSet;
+
+        public Repository(ApplicationDbContext context)
+        {
+            _context = context;
+            _dbSet = _context.Set<T>();
+        }
+        public void Delete(T entity)
+        {
+            _dbSet.Remove(entity);
+            _context.SaveChanges();
+        }
+
+        public IEnumerable<T> GetAll()
+        {
+            return _dbSet.ToList();
+        }
+
+        public T GetById(int? Id)
+        {
+            return _dbSet.Find(Id);
+        }
+
+        public void Insert(T entity)
+        {
+            _dbSet.Add(entity);
+            _context.SaveChanges();
+        }
+
+        public void Update(T entity)
+        {
+            _context.Update(entity);    
+            _context.SaveChanges();
```

Comment:
separate the SaveChanges() to the other method called Save()

Reply:
Should i do it only for this Repository.cs class or declare the method in IRepository.cs interface then implement it in the Repository.cs class?